/*
    Copyright 2002-2008 Tampere University of Technology.  All Rights
    Reserved.

    This file is part of TTA-Based Codesign Environment (TCE).

    TCE is free software; you can redistribute it and/or modify it under the
    terms of the GNU General Public License version 2 as published by the Free
    Software Foundation.

    TCE is distributed in the hope that it will be useful, but WITHOUT ANY
    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
    FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
    details.

    You should have received a copy of the GNU General Public License along
    with TCE; if not, write to the Free Software Foundation, Inc., 51 Franklin
    St, Fifth Floor, Boston, MA  02110-1301  USA

    As a special exception, you may use this file as part of a free software
    library without restriction.  Specifically, if other files instantiate
    templates or use macros or inline functions from this file, or you compile
    this file and link it with other files to produce an executable, this
    file does not by itself cause the resulting executable to be covered by
    the GNU General Public License.  This exception does not however
    invalidate any other reasons why the executable file might be covered by
    the GNU General Public License.
*/
/**
 * @file Exception.icc
 * @author Pekka J‰‰skel‰inen (pjaaskel-no.spam-cs.tut.fi) 2003
 *
 * Inline and template function definitions of exception classes.
 */

#include <string>


///////////////////////////////////////////////////////////////////////////////
// Exception
///////////////////////////////////////////////////////////////////////////////

/**
 * Returns the name of the file in which this exception occurred.
 */
inline std::string
Exception::fileName() const {
    return file_;
}

/**
 * Returns the line number of the file in which this exception occurred.
 */
inline int
Exception::lineNum() const {
    return line_;
}


/**
 * Returns the name of the procedure in which this exception was thrown.
 */
inline std::string
Exception::procedureName() const {
    return proc_;
}


///////////////////////////////////////////////////////////////////////////////
// PathNotFound
///////////////////////////////////////////////////////////////////////////////
inline std::string
PathNotFound::path() const {
    return path_;
}
